messaging/messaging-core/src/main/java/uk/gov/justice/services/messaging/DefaultJsonEnvelope.java

```diff
@@ -54,5 +58,50 @@ public JsonString payloadAsJsonString() {
         return (JsonString) payload;
     }
 
+    public static Builder envelope() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private JsonObjectMetadata.Builder metadata;
+        private final JsonObjectBuilderWrapper payload = new JsonObjectBuilderWrapper();
+
+        private Builder() {
+        }
+
+        public Builder with(final JsonObjectMetadata.Builder metadata) {
```

Comment:
typically you would be saying be saying
with(metadataOf(...) ... ) using the static metadataOf method  from the JsonObjectMetadata class. You get more fluent API that way.